backend/main/models/proposal.go

```diff
@@ -108,7 +108,14 @@ func (p *Proposal) GetProposalById(db *s.Database) error {
 	WHERE p.id = $1
 	GROUP BY p.id`
 	sql = fmt.Sprintf(sql, computedStatusSQL)
-	return pgxscan.Get(db.Context, db.Conn, p, sql, p.ID)
+	err := pgxscan.Get(db.Context, db.Conn, p, sql, p.ID)
+
+	if p.Max_weight == nil {
```

Comment:
I was thinking that if Max_weight isn't set, then there should be no Max_weight (i.e. just skip the check)